Core/RestaurantReservation.Application/Contracts/IRepository/IEmployeeRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using RestaurantReservation.Application.DTOs;
+using RestaurantReservation.Domain.Common;
+using RestaurantReservation.Domain.Enums;
+
+namespace RestaurantReservation.Application.Contracts.IRepository
+{
+    public interface IEmployeeRepository : IEntityRepository<EmployeeDTO>
+    {
+        Task<(IEnumerable<EmployeeDTO>, PaginationMetadata)> RetrieveAllAsync(
+            int pageNumber, int pageSize, EmployeePosition position);
```

Comment:
Same comment for checking out Seive or anyother way to filter on more than one field.